sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -245,10 +336,24 @@ namespace Azure { namespace Core {
       }
     }
 
-    /**
-     * @brief The application context (root).
+    /** @brief Creates a new root Context. This context is not a child of any other context, and
+     * thus can be independently cancelled.
+     *
+     * @return New root context.
+     *
+     */
+    static Context CreateNewRoot() { return Context{std::make_shared<ContextSharedState>()}; }
+
+    /** @brief The `ApplicationContext` is the root of all Context objects.
+     *
+     * @note Cancelling the `ApplicationContext` will cancel ALL operations currently active in the
+     * process, unless they were created with Azure::Core::Context::CreateNewRoot().
+     *
+     * @note If the `ApplicationContext` is cancelled, all subsequent operations will be cancelled.
+     * To reset the `ApplicationContext` to a non-cancelled state, use
```

Comment:
This comment needs to be adjusted based on the outcome from above. There is no way to reset the state. CreateNewRoot is creating a new disconnected root. ApplicationContext is forever unusable. 

Its the forever unusable statement that makes me wonder why we want ApplicationContext to have that behavior.